aws/src/main/java/org/apache/iceberg/aws/s3/BaseS3File.java

```diff
@@ -75,13 +76,14 @@ public boolean exists() {
     }
   }
 
-  protected HeadObjectResponse getObjectMetadata() throws S3Exception {
+  protected S3Object getObjectMetadata() throws S3Exception {
     if (metadata == null) {
-      HeadObjectRequest.Builder requestBuilder = HeadObjectRequest.builder()
+      ListObjectsV2Response response = client().listObjectsV2(ListObjectsV2Request.builder()
           .bucket(uri().bucket())
-          .key(uri().key());
-      S3RequestUtil.configureEncryption(awsProperties, requestBuilder);
-      metadata = client().headObject(requestBuilder.build());
+          .prefix(uri().key())
+          .maxKeys(1)
+          .build());
+      metadata = response.hasContents() ? response.contents().get(0) : null;
```

Comment:
I think this changes the exception behavior a little bit because now `S3InputFile.getLength()` will throw NPE as opposed to S3 NOT_FOUND in the event it does not exist and `getLength()` is called.  Not a huge issue, but we might consider improving that a little.

Reply:
Yeah I was going to ask, I saw in `HadoopInputFile` it simply throws a `RuntimeIOException` when asking for length of a file not exist. Probably better to check existence and throw `NotFoundException` in this case.

Reply:
I agree, `NotFoundException` would be much better.